src/Features/LanguageServer/Protocol/Handler/Diagnostics/DiagnosticSourceProviders/IDiagnosticSourceManager.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler.Diagnostics.DiagnosticSources;
+
+/// <summary>
+/// Manages the diagnostic sources that provide diagnostics for the language server.
```

Comment:
comment is lacking imo.  What does it mean to 'manage' here.  Maybe something more like thsi is responsible for aggregating disparate, IDiagnosticSourceProvider doing blah blah blah with them?